7_drivers_munge/src/GCM_driver_utils.R

```diff
@@ -242,26 +245,125 @@ download_gcm_data <- function(out_file_template, query_geom,
-  out_file <- sprintf(out_file_template, gcm_name, query_geom$tile_no[1])
+  out_file <- sprintf(out_file_template,
+                      gcm_name,
+                      gcm_projection_period,
+                      query_geom$tile_no[1])
 
   # write file
   arrow::write_feather(my_data, out_file)
 
   return(out_file)
 }
+
+#' @title Pull out individual file information from a dynamically mapped target
+#' @description Use a dynamically mapped target to extract information about
+#' the individual branch files to build a table that can group files and then
+#' trigger rebuilds when individual files within a group change.
+#' @param dynamic_branch_names character vector of the names of each branch created
+#' from running `names()` using the target dynamically branched target as input.
+#' @return a tibble with three columns (`name` = target name of the branch, `path`
+#' = character string of the file created by that branch, and `data` = hash code
+#' indicating the current state of the file) and a row for each branch.
+build_branch_file_hash_table <- function(dynamic_branch_names) {
+  # Get metadata for all the branches of the dynamic target
+  tar_meta(all_of(dynamic_branch_names), c("path", "data")) %>%
+    # There is just one path per branch, so remove the `list` format from this column
+    unnest(path)
+}
+
+#' @title Summarize hourly output from geoknife to daily
+#' @description the final GCM driver data will need to be daily, but geoknife
+#' returns hourly values. This step summarizes the data into daily values. It
+#' creates a file with the exact same name, except that the "_raw" part of the
+#' `in_file` filepath is replaced with "_daily".
+#' @param in_file filepath to a feather file containing the hourly geoknife data
+munge_to_glm <- function(in_file) {
+
+  daily_data <- arrow::read_feather(in_file) %>%
+
+    # Pivot to long format first to get cells as a column.
+    pivot_longer(cols = -c(DateTime, variable, statistic, units),
+                 names_to = "cell", values_to = "val") %>%
+    mutate(cell = as.numeric(cell)) %>%
+
+    # Now pivot wider to makes each variable a column for easy, readable munging
+    # Also, removes `units` and `statistic` columns which are specific to each variable
+    pivot_wider(id_cols = c("DateTime", "cell"), names_from = variable, values_from = val) %>%
+
+    # Unit conversions to get GLM-ready variables from GDP ones
+    mutate(AirTemp = tas - 273.15, # Convert GDP air temperature (tas) from Kelvin to Celcius
+           Rain = pr # TODO: Convert GDP precipitation (pr) from kg/m2/s to meters/hour
+    ) %>%
+
+    # Simply rename GDP variables into GLM variables
+    mutate(RelHum = qas,
+           Shortwave = rsns,
+           Longwave = rsns # TODO: find longwave from GDP
+    ) %>%
+
+    # Calculate GLM variables using other existing variables
+    mutate(Snow = ifelse(AirTemp < 0, Rain*10, 0), # When air temp is below freezing, any precip should be considered snow
```

Comment:
Just out of curiosity, @jread-usgs and @lindsayplatt, why is the snow value set to 10 times the rain when the temp is below 0? I see that's done [here](https://github.com/USGS-R/lake-temperature-model-prep/blob/main/7_drivers_munge/src/GLM_driver_utils.R#L153) too. Is 10 just a 'best guess' factor being used to account for the much lower density of snow?

Reply:
yes, 10 assumes snow depth is 10x the water equivalent. This varies depending on the type of snow, but is a good starting point. 

I also mentioned to Lindsay that we have the option of leaving the snow column blank, which then has GLM make this calculation internally.

Reply:
okay, thanks!